src/fusor/models.py

```diff
@@ -63,37 +62,34 @@ class FunctionalDomain(BaseModel):
 
     type: Literal[FUSORTypes.FUNCTIONAL_DOMAIN] = FUSORTypes.FUNCTIONAL_DOMAIN
     status: DomainStatus
-    associated_gene: GeneDescriptor
+    gene: Gene
     id: CURIE | None = Field(None, alias="_id")
     label: StrictStr | None = None
-    sequence_location: LocationDescriptor | None = None
-
-    _get_id_val = field_validator("id")(return_value)
```

Comment:
wanted to draw attention to these - it looks like everywhere `return_value` was used was verifying a CURIE. This is no longer needed. (And note to myself: I will make sure this is tested and works properly with the usage of CURIE from gene normalizer schema)